src/zos_files/zowe/zos_files_for_zowe_sdk/files.py

```diff
@@ -47,6 +47,18 @@ def __init__(self, connection):
         self.default_headers["Accept-Encoding"] = "gzip"
 
 
+    def __adjust_for_url(self, str_to_adjust):
+        """Adjust string to be correct in a URL
+
+        Returns
+        -------
+        adjusted_str
+            A string with special characters, acceptable for a URL
+        """
+
+        return str_to_adjust.replace("#", "%23") if str_to_adjust is not None else None
```

Comment:
@t1m0thyj thanks for the comment. I've tried other possible dataset names, looks like they are ok to use inside of URL (such as with @ and $ chars in them). But of course, it could be better to use the `urllib.parse.quote` to prevent further issues. Will do